cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -286,7 +313,7 @@ private Cluster tryReuseCluster(DataprocClient client, ProvisionerContext contex
           //Let's remove the reuse label to ensure new cluster will be picked up by findCluster
           try {
             client.updateClusterLabels(cluster.getName(), Collections.emptyMap(),
-                Collections.singleton(LABEL_RUN_KEY));
+                Collections.singleton(LABEL_RUN_KEY), context.getErrorCategory());
```

Comment:
nit: Consider passing error category to the client constructor through DataprocClientFactory#create. This way you won't need to provide it each time in each call. This error category is more like a context thing and seems a bit foreing/auxulary in specific calls